### ISSUE TITLE
Handle multiline expression in subject clause of if and case statements

### ISSUE THREAD

**Issue by avh4** [avh4/elm-format#17]
How to format something like the following:

```elm
foo a =
    if
        if a == Nothing then
            True
        else
            False
    then
        "Yes"
    else if
        if b == Nothing then
            True
        else
            False
    then
        "Perhaps"
    else
        "No"

bar a =
    case
        if a == Nothing then
            "X"
        else
            "Y"
    of
        _ -> ()
```

**Comment by rtfeldman** [comment]
The above seems like what we'd want, although I'd expand on it to clarify `else if`:

```elm
foo a =
    if
        if a == Nothing then
            True
        else
            False
    then
        "Yes"
    else if
        if b == Nothing then
            True
        else
            False
    then
        "Perhaps"
    else
        "No"
```

Delightful! :stuck_out_tongue_winking_eye:

**Comment by Apanatshka** [comment]
What @rtfeldman wrote seems good to me.

**Comment by avh4** [comment]
Okay, the agreement here seems pretty clear.  I've updated the description to include examples of `else if` and `case`.

**Comment by avh4** [comment]
Fixed in d79c4b6, d79c4b6^.
